saleor/checkout/base_calculations.py

```diff
@@ -4,16 +4,16 @@
 manager.
 """
 
-from typing import TYPE_CHECKING, List
+from typing import TYPE_CHECKING, Iterable, List, Optional
 
 from prices import TaxedMoney
 
 from ..core.taxes import quantize_price, zero_taxed_money
+from ..discount import DiscountInfo
```

Comment:
I see that you moved most of the imports of  `DiscountInfo`  to above `TYPE_CHECKING`.
Do we want to do that? IMO the safest way is to reduce import dependencies to minimal.

Reply:
`TYPE_CHECKING` is an ugly workaround for dependency loops, not the best practice for type imports 😉